### PR TITLE
chore: Migrates `mongodbatlas_cloud_backup_snapshot_export_job` to new auto-generated SDK

### PR DESCRIPTION
**Pull request by oarbusi**
## Description

Migrates `cloud_bakcup_snapshot_export_job` to new auto-generated SDK

Link to any related issue(s): CLOUDP-246318

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/service/cloudbackupsnapshotexportjob/data_source_cloud_backup_snapshot_export_job.go [hunk 1]
```diff
@@ -71,8 +71,9 @@ func DataSource() *schema.Resource {
 				Computed: true,
 			},
 			"err_msg": {
-				Type:     schema.TypeString,
-				Computed: true,
+				Type:       schema.TypeString,
+				Computed:   true,
+				Deprecated: fmt.Sprintf(constant.DeprecationParamByVersion, "1.18.0"),
```
Comment (marcosuma) [inline internal/service/cloudbackupsnapshotexportjob/data_source_cloud_backup_snapshot_export_job.go]: isn't 1.18.0 too soon? It's basically our next release which might contain this change itself.

**Review by EspenAlbert — APPROVED** [review]
LGTM! It would be nice to test the `components` with `export_id` and `replica_set_name` but that requires changing the cluster to sharded and is probably out of scope

File: internal/service/cloudbackupsnapshotexportjob/data_source_cloud_backup_snapshot_export_job.go (hunk 1 above)
Comment (oarbusi) [reply internal/service/cloudbackupsnapshotexportjob/data_source_cloud_backup_snapshot_export_job.go]: You are right! Changed to 1.20.0

**Review by carriecwk — APPROVED** [review]
LGTM

**Comment by marcosuma** [comment]
I think you have a mispell in the PR title: `mongodbatlas_cloud_bakcup_snapshot_export_job`

File: .changelog/2436.txt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+```release-note:note
+resource/mongodbatlas_cloud_backup_snapshot_export_job: Deprecates the `err_msg` attribute. The attribute is no longer returned from the backend and was considered low value for customers
```
Comment (lantoli) [inline .changelog/2436.txt]: i would remove the second sentence, i don't think we normally explain so much in detail in the changelog : "The attribute is no longer returned from the backend and was considered low value for customers"

File: docs/data-sources/cloud_backup_snapshot_export_job.md [hunk 3]
```diff
@@ -49,7 +49,7 @@ In addition to all arguments above, the following attributes are exported:
 * `custom_data` - Custom data to include in the metadata file named `.complete` that Atlas uploads to the bucket when the export job finishes. Custom data can be specified as key and value pairs.
 * `components` - _Returned for sharded clusters only._ Export job details for each replica set in the sharded cluster.
 * `created_at` - Timestamp in ISO 8601 date and time format in UTC when the export job was created.
-* `err_msg` - Error message, only if the export job failed.
+* `err_msg` - Error message, only if the export job failed. **Note:** This attribute is deprecated as it is not being used.
```
Comment (lantoli) [inline docs/data-sources/cloud_backup_snapshot_export_job.md:52]: i just saw here: https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2438
that we used a format like: **(DEPRECATED)** Use `replication_specs.#.region_config.#.

don't know which one is the best one, but we should unify

**Review by lantoli — APPROVED** [review]

**Pull request merged**